### PR TITLE
Functional Tests : API - Resource Endpoint

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Functional Tests : API - Resource Endpoint
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: & Nightly is :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/4383253669/ (The first part of the CI is locked due to #31675).
| Fixed ticket?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

### CODE REVIEW

**Review by nesrineabdmouleh — DISMISSED** [review]

File: tests/UI/utils/api.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  aud: string
+  jti: string
+  iat: number
+  nbf: number
+  exp: number
+  sub: string
+  scopes: string[]
+};
+
+function filterResponseHeader(headers: HttpHeader[], headerName: string): HttpHeader[] {
```
Comment (boubkerbribri) [inline tests/UI/utils/api.ts:31]: Need JsDoc

File: tests/UI/utils/api.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+}
+function parseJwtPayload(token: string): JwtAccessToken {
+  const base64Payload: string = token.split('.')[1];
+  const payload: Buffer = Buffer.from(base64Payload, 'base64');
+
+  return JSON.parse(payload.toString());
+}
+
+function extractAPIPrivateKey(): string {
+  const parametersFile: string = path.resolve(__dirname, '../../../', 'app/config/parameters.php');
```
Comment (boubkerbribri) [inline tests/UI/utils/api.ts]: i'm not sure that should be hard coded

File: tests/UI/utils/api.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/utils/api.ts:31]: Fixed

File: tests/UI/utils/api.ts (hunk 2 above)
Comment (Progi1984) [reply tests/UI/utils/api.ts]: FIxed in 1a6363a

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by nesrineabdmouleh** [comment]
Thank you @Progi1984

**Pull request merged**